File: lua/p004.lua
function isPalendromic(n)
	s = tostring(n)
	return s == string.reverse(s)
end
  
function main() 
	palendromes = {}
	for i = 100, 1000, 1 do
		for j = 100, 1000, 1 do
			prod = j * i 
			if isPalendromic(prod) then
				table.insert(palendromes, prod)
			end
		end
	end
	return math.max(unpack(palendromes))
end

print(main())

File: ruby/p004.rb
def isPalendromic(n)
	s = n.to_s()
	return s == s.reverse()
end

def main() 
	palendromes = []
	for i in 100..1000 do
		for j in 100..1000 do
			prod = j * i 
			if isPalendromic(prod) then
				palendromes.append(prod)
			end
		end
	end
	return palendromes.max()
end

puts main()